js/client-battle.js

```diff
@@ -1117,20 +1117,43 @@
 		toggleIgnoreSpects: function (e) {
 			this.battle.ignoreSpects = !!e.currentTarget.checked;
 			this.battle.add('Spectators ' + (this.battle.ignoreSpects ? '' : 'no longer ') + 'ignored.');
+			var $messages = $('.battle-log').find('[class*="chatmessage-"]').not('small:contains(\u2605)');
```

Comment:
I think that's not a problem. If you're ignoring spectators you probably don't want join/leave spam.